### DISCUSSION TITLE
"Pinched" Corners  |  Max X Probe  |  E-Steps

### DISCUSSION THREAD

**Discussion by poorandunlucky** [Jyers/Marlin#995] [Q&A] [answered]
First I just checked the files, and it looks like S-Curve acceleration control is enabled, which I didn't do tests with, and which sounds good on paper, but I was wondering if it's what could lead to my "pinched corners" on my test squares...  Square stuff comes out with the corners "pinched out".  Other sites talk about jerk control, and acceleration, and I have both at max in the firmware, and unset in Cura...  It's like the print head takes a break at the corners (visible when printing), and I don't know why.  Like L-moves aren't continuous, there's a perceptible amount of time the head spends on the corner before continuing the move, and I was wondering if the two could be related...

Second is, because of the tensioner on the right of the X-axis gantry on the Ender 3 V2, the print head can actually move farther to the right than all other Enders, to a point where the BLTouch could probe the rightmost part of the plate, or at least up to its margin, it seems...  When probing the right part of the plate, the BLTouch seems like it too far from the edge, or at least not as close to the edge as on the left, though it could definitely be maybe 1 cm farther than it currently seems to go.  The head itself is at the edge of the plate, but the probe is not quite there yet, and because of the tensioner, I think the carriage could make it there...  Is there a way to fix this?

Finally, I was wondering if other people had measured the extruder gear ratio to other than the firmware default of 93 steps/mm...  Once I measured it with calipers on another (stock?) firmware, and I got 94.3 or something, I measured it again when switching to this (Jyers) firmware, and 93 seemed dead-on, but because of the pinched corners, I'm wondering if my prints aren't under-extruded because of this...  I realize it's a gear with teeth, and it's outer circumference, to get 1 mm, you need 93 steps, on the stock machine with stock hardware, but I was still wondering if there was a link, and if anyone else had come-up with anything else than the defaults on stock hardware?

I haven't tried the firmwares since the ... whatchamaycallit...  the "leap"?  So I can't offer any feedback on that, but I do hope it's going well...

**Comment by smooreace** [comment]
In reality, the difference in 93 and 94.3 is so insignificant that its
likely completely unperceivable. with 100mm of material you lose a whopping
1.3mm of material at the higher setting.

**Reply by poorandunlucky** [reply]
Oh, I hadn't seen it that way...  Then my pinched corners are definitely not linked to under-extrusion during linear motion because of this...

To be honest, my measurement was pretty hard to take with the curvature of the filament, and how awkward it was to fit a caliper in there, and get stuff to stick to the filament without leaving adhesive behind (not to clog the nozzle or dirty the melting chamber)...  It was likely an error or something.  A better process on the 1.3.0 Jyers firmware showed me 93 steps/mm was dead-on, I had little variance, and experience in measuring it, so I'm more inclined to think 93 steps/mm is the good number, and my 94.3 was just a mistake I repeated twice...

Thanks for putting it in perspective like that for me, I appreciate...  I can really disregard that from now on...  Cleans-up a big part of the clutter in my head...

**Comment by smooreace** [comment]
I am currently dealing with some over extrusion on random corners too.
Print a flat panel. basically, 40mmX80mm at 3mm thick. I put my seams at
one corner, and it looks great. The strange part is that the OPPOSITE
corner has that bulging that you describe. The other 2 corners look great.
Still trying to narrow down exactly what's happening.

**Reply by poorandunlucky** [reply]
One website I looked at saw a reduction of the "pinching" by reducing the speed of the print, printing walls at around 25-30 mm/s gave them almost straight edges with almost imperceptible "pinching", but that's really, really slow...  It's not exactly feasible, or at least efficient to print a box enclosure at 25 mm/s, but I was able to correlate these results myself.

I also printed five (5) 25 x 18 x 7 mm blocks at various settings, and the best one was with the default 0.2 mm Cura settings.  I tried various jerks, acceleration, and things like that, and the best was with those features disabled, but as I increase the speed from the default, the pinching re-appears...

I'm not sure what does that, and it's not overly addressed, strangely...  Seems most videos on YouTube are of benchis, and other complex geometries (probably to attract attention/views), but I found little information about the pinching of square corners besides the relation with speed, where slower is better.

It doesn't explain the perceptible "wait time" at the corner, though, so maybe there's a problem in the way the board interprets the Gcode, like it processes the Gcode at the corner, and it takes time...

Maybe I could tell it to overextrude wall lines or something, calculate with calipers and try and figure out the difference, but I'm hoping someone already has a cause and solution figured out...

**Reply by gothcha** [reply]
Linear advance is intended to reduce this effect, there are threads on enabling it and the hardware  mods required see #707 
I enabled it on my 4.2.2, but I didn’t see a huge benefit on a test cube but larger prints at higher speeds could well benefit

**Comment by smooreace** [comment]
I'm REALLY tempted to swap to an SKR board and go this route. But a large
reason is to be able to properly run a dual-z.

**Comment by Jyers** [comment, marked as answer]
As @gothcha mentioned, linear advance is your best bet at reducing this. See #707

**Reply by poorandunlucky** [reply]
Even if I have a cheap logic analyzer/8 MHz USB scope, a collection of 74-series logic gates, a few MCUs, and spent the past few years impulse-buying passive components on Ali Express, I'm still not super comfortable making hardware mods to a $400 machine I could scrap bodging...

I'm just not confident enough, both in my skills, my ability to understand and follow directions, and my ability to piece all the information together yet, and I'm afraid of scraping my board, plus I'm kinda depressed right now I think, I don't have a lot of energy...

I just remembered the setting from when I tried to configure Marlin on my own, and was about to buy a LCD screen for a normal Ender 3 (just before finding your firmware), and I was wondering if it could be related, somehow turning a square into a spline or something... idk...

But yeah, I'll think about it, and when I get sufficiently fed-up, and have enough courage, I'll pick the mainboard out, I want to change the fans, too, despite the silent steppers, the thing is really f-in loud...

I was really hoping for a turnkey solution, to be honest, i'm pretty disappointed with the Ender 3 V2...  Everywhere on YouTube "it's a great starting machine" and indeed it's better than the first one I returned to Amazon that didn't have the silent steppers (omg i thought I was in a car factory was sooo f-in LOUD!!!), but I think they could better...  It could be worse though, there's enough people who have one, and there's you.  Your really save the day, let me tell you...

I haven't installed the releases past 1.3.2 I think, before you started switching to the other display module to get your code upstream, but I hope that's going really well for you, and I hope you get rewarded, somehow, for this work you're doing...  I think it's really amazing, you're really, literally, saving my machine and my hopes towards 3D printing...

I don't think I could ever thank you enough for making the advanced features work on the DWIN display...

You've honestly restored my willingness to continue doing the 3D printing stuff...

But back on topic, if anyone is having this problem and stumbles on this in the future, aside the hardware mods, changing the wall speed to something very low has shown decent, or at least tolerable results.

I also got better results with the default Cura 0.2 mm height print settings than my own i had started developing, so anyone starting out, it might be worth it going back to the base, and going back up from there...

**Reply by Jyers** [reply]
I can certainly understand such hesitation, especially with such a mod. Because even though it's rather simple and doesn't require much material it is still some fairly fine soldering work.

The fan's aren't too hard of a change, most are plug and play. The power supply is the big one, and also the loud one... But with some patience and careful disassembly as well as some optional dremel work, you can get a very much improved noise profile for the printer.

I'm glad my work has been able to help you in your 3d printing endeavors and I wish you the best in your future progress :)

PS - Cura has a linear advance-ish type feature called coasting. No hardware mod needed. Might be worth a try.

**Reply by smooreace** [reply]
Realistically... you would be soldering on a $40 board. The rest of the generic and easily replaced parts have nothing to do with the mod work itself

**Reply by Reggi3** [reply]
You should be looking at a number of things really to solve bulging etc. not all of them will work but they will reduce the issues.  As you've said, Speed is something but also retraction/un-retraction, acceleration and jerk tuning in the firmware, combing, wiping and coasting in the slicer.  Make sure your flow is also reasonably well tuned.

Lastly for anyone that's thinking about linear advance and the advantages it gives or you're just having issues with corners etc., I would suggest you look at [the marlin linear advance page](https://marlinfw.org/docs/features/lin_advance.html) specifically the section headed 'Evaluating the Calibration Pattern' to get an idea of what is actually happening on a print move/series of moves.

I am by no means an expert, nor am I claiming linear advance fixes everything but understanding what it's attempting to fix will help both sets of people, those who want linear advance and those who just want a stock firmware to print a bit more consistently.

Lastly, if you've got all the kit but still feel hesitant at modding, practice on some old tech, your confidence will grow along with your faith in your own abilities, as far as doing this kind of thing is concerned, we all started at the same place, buying an iron, with the same skill set (none)...
